File: java/FullTimeInstructor.java
import java.util.Date;
import java.util.List;

public class FullTimeInstructor extends FullTimeEmployee {
    private String rank;
    private List<String> coursesTaught;

    public FullTimeInstructor(String id, String firstName, String lastName, Date dob,
            String position, String department, double baseSalary, String benefits,
            String rank, List<String> courses) throws InvalidDataException {
        super(id, firstName, lastName, dob, position, department, baseSalary, benefits);

        // Subclass-specific validation
        if (!"Staff".equals(department)) {
            throw new InvalidDataException("FullTimeInstructor must belong to the Staff department.");
        }

        // Only allow full-time positions
        if ("Adjunct Instructor".equals(position)) {
            throw new InvalidDataException("Adjunct Instructor cannot be full-time; use PartTimeInstructor instead.");
        }

        this.rank = rank;
        this.coursesTaught = courses;
    }

    @Override
    public void reportToManager() {
        if (manager != null) {
            System.out.println(getFullName() + " (" + rank + ", Full-Time Instructor) reports to manager "
                    + manager.getFullName());
        } else {
            System.out.println(getFullName() + " (" + rank + ", Full-Time Instructor) has no assigned manager.");
        }
    }

    public void submitGrades() {
        System.out.println(getFullName() + " (" + rank + ") submits grades.");
    }

    public void giveOfficeHours() {
        System.out.println(getFullName() + " (" + rank + ") gives office hours.");
    }

    public void showCoursesTaught() {
        System.out.println(getFullName() + " (" + rank + ") teaches:");

        if (coursesTaught == null || coursesTaught.isEmpty()) {
            System.out.println("  No courses assigned.");
            return;
        }

        for (String course : coursesTaught) {
            System.out.println("  - " + course);
        }
    }
}

File: csharp/FullTimeInstructor.cs
using System;
using System.Collections.Generic;

[Serializable]
public class FullTimeInstructor : FullTimeEmployee
{
    private string Rank { get; set; }
    private List<string> CoursesTaught { get; set; }

    public FullTimeInstructor() : base() { }

    public FullTimeInstructor(string id, string firstName, string lastName, DateTime dob,
        string position, string department, double baseSalary, string benefits,
        string rank, List<string> courses)
        : base(id, firstName, lastName, dob, position, department, baseSalary, benefits)
    {
        if (department != "Staff")
            throw new InvalidDataException("FullTimeInstructor must belong to the Staff department.");
        if (position == "Adjunct Instructor")
            throw new InvalidDataException("Adjunct Instructor cannot be full-time; use PartTimeInstructor instead.");

        Rank = rank;
        CoursesTaught = courses;
    }

    public override void ReportToManager()
    {
        if (Manager != null)
            Console.WriteLine($"{GetFullName()} ({Rank}, Full-Time Instructor) reports to manager {Manager.GetFullName()}");
        else
            Console.WriteLine($"{GetFullName()} ({Rank}, Full-Time Instructor) has no assigned manager.");
    }

    public void SubmitGrades() => Console.WriteLine($"{GetFullName()} ({Rank}) submits grades.");
    public void GiveOfficeHours() => Console.WriteLine($"{GetFullName()} ({Rank}) gives office hours.");
    public void ShowCoursesTaught()
    {
        Console.WriteLine($"{GetFullName()} ({Rank}) teaches:");
        if (CoursesTaught == null || CoursesTaught.Count == 0)
            Console.WriteLine("  No courses assigned.");
        else
            CoursesTaught.ForEach(c => Console.WriteLine($"  - {c}"));
    }
}
